picoCTF-shell/hacksport/deploy.py

```diff
@@ -65,19 +75,21 @@ def give_port():
         return inv_port_map[(context["problem"], context["instance"])]
 
     if len(context["port_map"].items()) + len(
-            context["config"].banned_ports_parsed) == 65536:
+            context["config"].banned_ports_parsed) == HIGHEST_PORT + 1:
         raise Exception(
             "All usable ports are taken. Cannot deploy any more instances.")
 
-    while True:
-        port = port_random.randint(0, 65535)
-        if port not in context["config"].banned_ports_parsed:
-            owner, instance = context["port_map"].get(port, (None, None))
-            if owner is None or (owner == context["problem"] and
-                                 instance == context["instance"]):
-                context["port_map"][port] = (context["problem"],
-                                             context["instance"])
-                return port
+    # collect list of banned ports
+    banned_ports_list = []
+    for port in port_map:
+        banned_ports_list.append(port)
+
+    # Get a random port that is not in the banned list. 
+    port = port_random.choice([i for i in range(LOWEST_PORT, HIGHEST_PORT)
+        if i not in banned_ports_list and not check_if_port_in_use(i)])
```

Comment:
I think @carlislemc is right, the list comprehension seems like it would run `check_if_port_in_use` on every single port in process of building that list.

Reply:
10 seems low, but maybe just throw an exception if you try 100 and it doesn't work.